### DISCUSSION TITLE
Invalid certificate: Error: Invalid certificate.

### DISCUSSION THREAD

**Discussion by ariqmbrilian** [k3s-io/k3s#9096] [Q&A] [answered]
I have single node k3s cluster version v1.28.4+k3s2. I want to connect it to Azure DevOps service connection but show error like the title. I have generated new kubeconfig with new public ip  curl -sfL https://get.k3s.io | INSTALL_K3S_VERSION=v1.28.4+k3s2 INSTALL_K3S_EXEC="--tls-san public ip" sh -
![image](https://github.com/k3s-io/k3s/assets/61818849/c60fc783-5887-4d73-9364-f820ef1cd85b)

**Comment by brandond** [comment, marked as answer]
I suspect you need to modify the k3s kubeconfig to point at the public IP of your node, as described at: https://docs.k3s.io/cluster-access#accessing-the-cluster-from-outside-with-kubectl

**Comment by ariqmbrilian** [comment]
i have change from 127.0.0.0 or localhost to my public ip. i also generate kubeconfig with SAN my public ip

**Reply by brandond** [reply]
Did you use the correct port as well?

**Reply by ariqmbrilian** [reply]
yes

**Reply by brandond** [reply]
Either that's the wrong kubeconfig or the URL is wrong. The CA certificates are embedded in the kubeconfig itself so I'm not sure how you would be getting a validation error.

**Comment by getangar** [comment]
I have the same issue and the kubeconfig is correct: I can interact with the cluster by using kubectl without issues. Just ADO won't work.

**Reply by brandond** [reply]
Maybe open an issue with them then?

**Reply by ariqmbrilian** [reply]
Have you try to create new service account? The problem is solved by myself
